proto/sentry_protos/snuba/v1/trace_item_attribute.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package sentry_protos.snuba.v1;
+
+message AttributeKey {
+  enum Type { //this mostly reflects what types are able to be ingested, see eap_spans consumer for ingest details
+    TYPE_UNSPECIFIED = 0; //called "none" sometimes
+    TYPE_BOOLEAN = 1;
+    TYPE_STRING = 2;
+    TYPE_FLOAT = 3;
+    TYPE_INT = 4; //note: all numbers are stored as float64, so massive integers can be rounded. USE STRING FOR IDS.
+  }
+
+  Type type = 1;
+
+  //if you use span_id this will route to span.span_id,
+  //if you use derp this will route to attr_{str,num}_{n}['derp']
+  string name = 2;
+}
+
+// custom mappings of column values which are not present in EAP storage
+// for example, project name is changeable by the user and not stored in EAP
+// but sorting by the project name may be desired. This struct carries that mapping
+// data
+message VirtualColumnContext {
+  string from_column_name = 1;
+  string to_column_name = 2;
+  map<string, string> value_map = 3;
+}
+
+message AttributeValue {
```

Comment:
Do we expect the `AttributeValue` ever to be `null`?
I am not very familiar with protobufs, but we might need to use wrappers or optional keyword here if we want to have null as a value?

Reply:
Good point